packages/flutter/test/material/theme_test.dart

```diff
@@ -265,4 +265,40 @@ void main() {
     expect(glyphText.text.style.color, Colors.orange);
     expect(glyphText.text.style.fontSize, 20.0);
   });
+
+  testWidgets(
+    'Same ThemeData reapplied does not trigger didChangeDepedencies',
+    (WidgetTester tester) async {
+      testBuildCalled = 0;
+      final ThemeData themeData = new ThemeData(primaryColor: const Color(0xFF000000));
+
+      await tester.pumpWidget(
+        new MaterialApp(
+          theme: themeData,
+          home: new Test(),
+        ),
+      );
+      expect(testBuildCalled, 1);
+      tester.element(find.byType(MaterialApp)).markNeedsBuild();
+      await tester.pump();
+
+      // No repeated build calls to the child since it's the same theme data.
+      expect(testBuildCalled, 1);
+    },
+  );
+}
+
+int testBuildCalled;
+class Test extends StatefulWidget {
+  @override
+  _TestState createState() => new _TestState();
+}
+
+class _TestState extends State<Test> {
+
+  @override
+  Widget build(BuildContext context) {
+    ++testBuildCalled;
```

Comment:
fwiw, and you should feel free to ignore me here if you disagree, i find `testBuildCalled += 1` reads slightly better than using the `++` prefix operator. (then again, i'd remove the `++` operator from the language entirely if i could!)